hub/core/storage/memory.py

```diff
@@ -1,19 +1,86 @@
-from hub.core.storage.mapped_provider import MappedProvider
-import fsspec  # type: ignore
+from hub.core.storage.provider import StorageProvider
 
 
-class MemoryProvider(MappedProvider):
+class MemoryProvider(StorageProvider):
     """Provider class for using the memory."""
 
     def __init__(self, root):
-        """Initializes the MemoryProvider.
+        self.mapper = {}
+
+    def __getitem__(
+        self,
+        path: str,
+    ):
+        """Gets the object present at the path within the given byte range.
+
+        Example:
+            memory_provider = MemoryProvider("xyz")
```

Comment:
I don't think memory provider needs to have a root.

Reply:
the API should be as uniform as possible, & i think it is nice to have root because if we have multiple workers using the `MemoryProvider`, presumably they can be using different roots.

it should seem exactly like a file system